### ISSUE TITLE
A question about the testing process.

### ISSUE THREAD

**Issue by ghost** [Songweiping/GRU4Rec_TensorFlow#6]
Hello, Song,

I found a problem in this kind of process.
In your evaluation.py:
![image](https://user-images.githubusercontent.com/26358826/36069167-936b433c-0f1f-11e8-9694-bad1f7be0437.png)
I note that you picked  ">" here, but if we use tanh as the final activation function, many scores of this preds may be equal to 1 for each user. And that means many scores of items are in one position at the same time.  But this trick is unreasonable, because we cannot rank these items.  If you use >=, the performance will be bad.

I think this is the implement in the original Hidash's code.  So what do you think of this? 

Looking forward to your reply.
Thank you ~

**Comment by Songweiping** [comment]
Thanks for discussing this!
You are right this is the implement in the original Hidash's code. While I agree that using tanh as final activation function maybe problematic, especially together with cross entropy loss (negative number in log()). 
I only run this code with softmax activation and cross entropy loss. Would you please share a bit more on how you use tanh in your experiments? With what kind of loss function? How the performance is going?

**Comment by iejiaochenhao** [comment]
The original paper said that it is better to use tanh.
![1557557837(1)](https://user-images.githubusercontent.com/45477969/57566309-82ce7780-73fd-11e9-8565-2c8032a977f0.png)
